features/nanostack/targets/TARGET_NXP/TARGET_KW41Z/NanostackRfPhyKw41z.cpp

```diff
@@ -0,0 +1,1093 @@
+            rf_receive();
+        } else {
+            PhyIsrSeqCleanup();
+            switch(xcvseqCopy)
+            {
+            case gTX_c:
+                if ((ZLL->PHY_CTRL & ZLL_PHY_CTRL_CCABFRTX_MASK) &&
+                    (irqStatus & ZLL_IRQSTS_CCA_MASK)) {
+                    device_driver.phy_tx_done_cb(rf_radio_driver_id, rf_mac_handle,
+                                                 PHY_LINK_CCA_FAIL, 1, 1);
+                } else {
+                    rf_handle_tx_end(false);
+                }
+                break;
+
+            case gTR_c:
+                if ((ZLL->PHY_CTRL & ZLL_PHY_CTRL_CCABFRTX_MASK) &&
+                    (irqStatus & ZLL_IRQSTS_CCA_MASK)) {
+                    device_driver.phy_tx_done_cb(rf_radio_driver_id, rf_mac_handle,
+                                                 PHY_LINK_CCA_FAIL, 1, 1);
+                } else {
+                    rf_handle_tx_end((irqStatus & ZLL_IRQSTS_RX_FRM_PEND_MASK) > 0);
+                }
+                break;
+
+            case gRX_c:
+                rf_handle_rx_end();
+                break;
+
+            case gCCA_c:
+                rf_ed_value = rf_convert_energy_level((ZLL->LQI_AND_RSSI &
+                                                       ZLL_LQI_AND_RSSI_CCA1_ED_FNL_MASK) >>
+                                                       ZLL_LQI_AND_RSSI_CCA1_ED_FNL_SHIFT);
+                break;
+
+            default:
+                break;
+            }
+        }
+    }
+}
+
+NanostackRfPhyKw41z::NanostackRfPhyKw41z()
+{
+    memcpy(MAC64_addr, MAC64_addr_default, sizeof(MAC64_addr));
+}
+
+NanostackRfPhyKw41z::~NanostackRfPhyKw41z()
+{
+    // Do nothing
+}
+
+int8_t NanostackRfPhyKw41z::rf_register()
+{
+    platform_enter_critical();
+
+    if (rf != NULL) {
+        platform_exit_critical();
+        error("Multiple registrations of NanostackRfPhyKw41z not supported");
+        return -1;
+    }
+
+    rf = this;
+    int8_t radio_id = rf_device_register();
+    if (radio_id < 0) {
+        rf = NULL;
+    }
+
+    platform_exit_critical();
+    return radio_id;
+}
+
+void NanostackRfPhyKw41z::rf_unregister()
+{
+    platform_enter_critical();
+
+    if (rf != this) {
+        platform_exit_critical();
+        return;
+    }
+
+    rf_device_unregister();
+    rf = NULL;
+
+    platform_exit_critical();
+}
+
+void NanostackRfPhyKw41z::get_mac_address(uint8_t *mac)
+{
+    platform_enter_critical();
+
+    memcpy((void*)mac, (void*)MAC64_addr, sizeof(MAC64_addr));
+
+    platform_exit_critical();
+}
+
+void NanostackRfPhyKw41z::set_mac_address(uint8_t *mac)
+{
+    platform_enter_critical();
+
+    if (NULL != rf) {
+        error("NanostackRfPhyKw41z cannot change mac address when running");
+        platform_exit_critical();
+        return;
+    }
+    memcpy((void*)MAC64_addr, (void*)mac, sizeof(MAC64_addr));
+
+    platform_exit_critical();
+}
+
+NanostackRfPhy &NanostackRfPhy::get_default_instance()
```

Comment:
I think this should really be `MBED_WEAK` - for the PHYs we have the model that there's no default implementation in the OS, the target can supply a weak implementation, and the app or a plug-in module can provide normal.

This allows the theoretical possibility of plugging in a phy module into a board that has a built-in phy.

Can be fixed later though.